src/app/3d/qgs3dmapcanvas.cpp

```diff
@@ -61,7 +61,7 @@ Qgs3DMapCanvas::~Qgs3DMapCanvas()
   // make sure the scene is deleted while map settings object is still alive
   delete mScene;
   mScene = nullptr;
-
+  emit canvasClosing();
```

Comment:
I think if you put something like
```
if ( mMapTool )
  mMapTool->deactivate();
```
at the top of the destructor body (before the scene is deleted) you don't need any signal about canvas getting closed...

Reply:
You're right, this was an oversight on my end.